### ISSUE TITLE
bash disabled or not supported with --local

### ISSUE THREAD

**Issue by ekg** [KillianLucas/open-interpreter#829]
### Describe the bug

Using LMStudio with TheBloke/OpenHermes-2.5-Mistral-7B-16k-GGUF, I get:

```
> list the files in the current directory


                                                                                                                                                               
  ls                                                                                                                                                           
                                                                                                                                                               
                                                                                                                                                               
  `bash` disabled or not supported.                                                                                                                            
                                                                                                                                                               
```

How can I enable bash for --local?

### Reproduce

Start up LMStudio, run the server.

`interpreter --local -safe off`


### Expected behavior

The command should be run.

### Screenshots

_No response_

### Open Interpreter version

0.2.0

### Python version

3.10.12

### Operating System name and version

Ubuntu 22.04

### Additional context

_No response_

**Comment by ekg** [comment]
Any suggestions? Just a word or two about where to hack would speed me on my way to resolve this.

**Comment by SunnyOd** [comment]
Getting the same error here too. I'm using `ollama` with `litellm` and `code-llama`

**Comment by Notnaton** [comment]
https://github.com/KillianLucas/open-interpreter/tree/main/interpreter%2Fcore%2Fcomputer%2Fterminal%2Flanguages

![Screenshot_20231229-004826.png](https://github.com/KillianLucas/open-interpreter/assets/4324290/6ecc2155-9f57-4d53-adaa-cb0123263600)

Instead of bash, make it use shell script as bash is not supported.

**Comment by SunnyOd** [comment]
Thanks @Notnaton. I normally run `interpreter` via a shell script so it sets up the conda environment, etc. The script was prefixed with `#!/bin/bash` before so I changed it to `#!/bin/sh` and tried running the script again but it still fails with:
```
`bash` disabled or not supported.   
`python` disabled or not supported. 
```

Am I setting shell up correctly?

Thanks!
S

**Comment by Notnaton** [comment]
Can you run `python` in the command line environment?

**Comment by cavit99** [comment]
I'm having the exact same issue.

**Comment by mexicanamerican** [comment]
> I'm having the exact same issue.



> ### Describe the bug
> Using LMStudio with TheBloke/OpenHermes-2.5-Mistral-7B-16k-GGUF, I get:
> 
> ```
> > list the files in the current directory
> 
> 
>                                                                                                                                                                
>   ls                                                                                                                                                           
>                                                                                                                                                                
>                                                                                                                                                                
>   `bash` disabled or not supported.                                                                                                                            
>                                                                                                                                                                
> ```
> 
> How can I enable bash for --local?
> 
> ### Reproduce
> Start up LMStudio, run the server.
> 
> `interpreter --local -safe off`
> 
> ### Expected behavior
> The command should be run.
> 
> ### Screenshots
> _No response_
> 
> ### Open Interpreter version
> 0.2.0
> 
> ### Python version
> 3.10.12
> 
> ### Operating System name and version
> Ubuntu 22.04
> 
> ### Additional context
> _No response_

Having the same issue

**Comment by Speedway1** [comment]
Same here. Got the same issue.

**Comment by Speedway1** [comment]
I have solved the bug and raised bug ticket #867. However, for those in a rush, here is a quick and dirty fix until the patch/bug fix is applied:


1) go to your virtual environment where the "pip install" has installed the code (e.g. open_interpreter_env)
2) edit the "respond.py" file, for example:
      vi open_interpreter_env/lib/python3.10/site-packages/interpreter/core/respond.py
3) change the code that looks like this:
                # Is this language enabled/supported?
                if language not in [
                    i.name.lower() for i in interpreter.computer.terminal.languages
                ]:
                    output = f"`{language}` disabled or not supported."
TO LOOK LIKE THIS:
                # Is this language enabled/supported?
                if language not in [
                    i.name.lower() for i in interpreter.computer.terminal.languages
                ] and language != "bash":
                    output = f"`{language}` disabled or not supported."

i.e. you are changing this line:
                ]:
to this:
                ] and language != "bash":
4) save (ZZ) and then restart "interpreter" and now your bash shell commands should work.

**Comment by SunnyOd** [comment]
Awesome news😎 Thank you!
